DashAI/back/models/hugging_face/opus_mt_en_es_transformer.py

```diff
@@ -0,0 +1,151 @@
+import shutil
+from typing import List
+
+from sklearn.exceptions import NotFittedError
+from transformers import (
+    AutoModelForSeq2SeqLM,
+    AutoTokenizer,
+    Seq2SeqTrainer,
+    Seq2SeqTrainingArguments,
+)
+
+from DashAI.back.dataloaders.classes.dashai_dataset import DashAIDataset
+from DashAI.back.models.translation_model import TranslationModel
+
+
+class OpusMtEnESTransformer(TranslationModel):
+    """
+    Pre-trained transformer opus-mt-en-es allowing to translate english texts
+    to spanish.
+    """
+
+    def __init__(self, model=None, **kwargs):
+        """
+        Initialize the transformer class by calling the pretrained model and its
+        tokenizer. Include an attribute analogous to sklearn's check_is_fitted to
+        see if it was fine-tuned.
+        """
+        self.model_name = "Helsinki-NLP/opus-mt-en-es"
+        self.tokenizer = AutoTokenizer.from_pretrained(self.model_name)
+        self.model = (
+            model
+            if model is not None
+            else AutoModelForSeq2SeqLM.from_pretrained(self.model_name)
+        )
+        self.fitted = model is not None
+        if model is None:
+            self.training_args = kwargs
+            self.batch_size = kwargs.pop("batch_size")
+            self.device = kwargs.pop("device")
+
+    def fit(self, dataset: DashAIDataset):
+        """Fine-tune the pre-trained model.
+
+        Parameters
+        ----------
+        dataset : DashAIDataset
+            DashAIDataset with training data.
+
+        """
+
+        input_column = dataset.inputs_columns[0]
+        output_column = dataset.outputs_columns[0]
+
+        def tokenize(examples):
+            inputs = self.tokenizer(
+                examples[input_column],
+                truncation=True,
+                padding="max_length",
+                max_length=512,
+            )
+            outputs = self.tokenizer(
+                examples[output_column],
+                truncation=True,
+                padding="max_length",
+                max_length=512,
+            )
+            inputs["labels"] = outputs["input_ids"]
+            return inputs
+
+        dataset = dataset.map(tokenize, batched=True)
+        dataset.set_format("torch", columns=["input_ids", "attention_mask", "labels"])
+
+        # Arguments for fine-tuning
+        training_args = Seq2SeqTrainingArguments(
+            output_dir="DashAI/back/user_models/temp_checkpoints_opus-mt-en-es",
+            save_steps=1,
+            save_total_limit=1,
+            per_device_train_batch_size=self.batch_size,
+            per_device_eval_batch_size=self.batch_size,
+            no_cuda=self.device != "gpu",
+            **self.training_args,
```

Comment:
Why not just get `self.batch_size` and `self.device` from `self.training_params` instead of storing them as instance attributes in `__init__()`?

Reply:
Or maybe the reverse, instantiate as instance variable the `Seq2SeqTrainingArguments` in the `__init__` as `self.training_args = Seq2SeqTrainingArguments(...)` (similar to how you do it with the tokenizer)?